### PR TITLE
Swap the concrete limiter to interface and add WithCustomLimiter

### LINKED ISSUE #909: Add support for custom rate limiter
Sort of a feature request, use cases may include private gitlab instance not doing the rate limiting itself and not returning rate limit headers and / or when someone wishes to limit load to lower than what's automatically configured. 

Can make a pull request to add it, wanted some thoughts and considerations that I'm not thinking about :)

Thanks.

### PR DESCRIPTION
**Pull request by 2785**
closes #909

### CODE REVIEW

File: client_options.go [hunk 1]
```diff
@@ -47,3 +47,13 @@ func WithoutRetries() ClientOptionFunc {
 		return nil
 	}
 }
+
+// WithCustomLimiter injects a custom rate limiter to the client.
+func WithCustomLimiter(limiter RateLimiter) ClientOptionFunc {
```
Comment (svanharmelen) [inline client_options.go:52]: I try to keep these options ordered in alphabetical order, so if you could update that...

File: gitlab.go [hunk 2]
```diff
@@ -185,6 +185,11 @@ type ListOptions struct {
 	PerPage int `url:"per_page,omitempty" json:"per_page,omitempty"`
 }
 
+// RateLimiter specifies a dependency that can wait on a context.
```
Comment (svanharmelen) [inline gitlab.go:188]:
```suggestion
// RateLimiter describes the interface that all (custom) rate limiters must implement.
```

File: client_options.go (hunk 1 above)
Comment (2785) [reply client_options.go:52]: o yeah I can do that.

Comment (2785) [reply client_options.go]: done.

**Pull request merged**